src/bentoml/_internal/service/openapi/utils.py

```diff
@@ -43,6 +52,33 @@ def pydantic_components_schema(pydantic_model: t.Type[pydantic.BaseModel]):
     return {k: Schema(**definitions[k]) for k in sorted(definitions)}
 
 
+def typeddict_components_schema(typeddict: t.Type[TypedDict]):
+    typeddict_set = get_flat_typeddicts_from_typeddict(
+        typeddict=typeddict, typeddict_set=set()
+    )
+    definitions: dict[str, Schema] = {}
+    for typeddict in typeddict_set:
+        definitions[typeddict.__name__] = typed_dict_to_dict(typeddict)
```

Comment:
```suggestion
        definitions[typeddict.__name__] = typed_dict_to_dict(typeddict)
```

This actually doesn't handle the case where there are two typeddict, `user.Model` and `base.Model`. One of them will just overwrite the other.

Reply:
You probably need to handle conflicting nametype with something like `f'{model.__module__}__{model.__qualname__}'.replace('.', '__')`, and provide a mapping to it.